x-pack/plugin/ilm/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ilm/actions/SearchableSnapshotActionIT.java

```diff
@@ -851,14 +851,13 @@ public void testSearchableSnapshotsInHotPhasePinnedToHotNodes() throws Exception
             )
         );
 
+        // rollover the data stream so searchable_snapshot can complete
         indexDocument(client(), dataStream, true);
-        String firstGenIndex = DataStream.getDefaultBackingIndexName(dataStream, 1L);
+        var backingIndices = getBackingIndices(client(), dataStream);
+        String firstGenIndex = backingIndices.get(0);
         Map<String, Object> indexSettings = getIndexSettingsAsMap(firstGenIndex);
         assertThat(indexSettings.get(DataTier.TIER_PREFERENCE), is("data_hot"));
```

Comment:
Should we check this before we index the document? The chances it will rollover so quickly is unlikely but better to be sure right?

Reply:
Ah yeah I think I thought about that but concluded that this was fine as `backingIndices.get(0)` would point to the same index before and after rollover, but I forgot to take into account that the searchable snapshot action removes the original index, of course. Good suggestion, will do :+1:

Reply:
Ah no nvm, I remember why I didn't move the check before the document indexation; the data stream doesn't exist yet before the document is indexed :smile:. I'll move the check and add a `PUT _data_stream/...` request.